File: 백준/Silver/11568. 민균이의 계략/민균이의 계략.kt
import kotlin.math.max
fun main()
{
    var arr = Array (1000){0}
    var table = Array (1000){0}
    val N = readln().toInt()
    val a=readln().split(" ")
    var answer:Int =0
    for (i in 0..<N)
    {
        arr[i]=a[i].toInt()
        var temp:Int =0
        for (j in 0..<i)
        {
            if(arr[j]<arr[i])
            {
                temp=max(temp,table[j])
            }
        }
        table[i]=temp+1
        answer=max(answer,table[i])
    }
    print(answer)
}

File: 백준/Silver/11568. 민균이의 계략/민균이의 계략.cc
#include <iostream>
using namespace std;
int nums[1000];
int table[1000];
int main(void)
{
	ios::sync_with_stdio(0); cin.tie(0); cout.tie(0);
	int N,answer=0;
	cin >> N;
	for (int i = 0; i < N; i++)
	{
		cin >> nums[i];
		int maxx = 0;
		for (int j = 0; j < i; j++)
		{
			if (nums[j] < nums[i])
			{
				maxx = max(maxx, table[j]);
			}
		}
		table[i] = maxx + 1;
		answer = max(answer, table[i]);
	}
	cout << answer;
	return 0;
}